api/base/serializers.py

```diff
@@ -177,8 +234,27 @@ def to_representation(self, obj, envelope='data'):
         meta = getattr(self, 'Meta', None)
         type_ = getattr(meta, 'type_', None)
         assert type_ is not None, 'Must define Meta.type_'
-        data = super(JSONAPISerializer, self).to_representation(obj)
-        data['type'] = type_
+
+        data = collections.OrderedDict([('id', ''), ('type', type_), ('attributes', collections.OrderedDict()),
+                                        ('relationships', collections.OrderedDict()), ('links', {})])
+
+        fields = [field for field in self.fields.values() if not field.write_only]
+
+        for field in fields:
+            try:
+                attribute = field.get_attribute(obj)
+            except SkipField:
+                continue
+
+            if isinstance(field, HyperlinkedIdentityFieldWithMeta):
```

Comment:
That's how I originally wrote it, but I was trying to force people to use HyperlinkedIdentityFieldWithMeta to get the formatting correct.  Below, `parent` is now using HyperlinkedIdentityField and this doesn't adhere to the spec.

![screen shot 2015-08-28 at 4 00 02 pm](https://cloud.githubusercontent.com/assets/9755598/9555765/7a6680f4-4d9d-11e5-80de-efd2d9133065.png)